apps/explorer/lib/explorer/exchange_rates/source/mobula.ex

```diff
@@ -23,20 +24,42 @@ defmodule Explorer.ExchangeRates.Source.Mobula do
       %Token{
         available_supply: to_decimal(market_data["circulating_supply"]),
         total_supply: to_decimal(market_data["total_supply"]) || to_decimal(market_data["circulating_supply"]),
-        btc_value: btc_value,
+        btc_value: to_decimal(btc_value),
         id: id,
         last_updated: nil,
         market_cap_usd: to_decimal(market_data["market_cap"]),
         tvl_usd: nil,
         name: market_data["name"],
         symbol: String.upcase(market_data["symbol"]),
-        usd_value: current_price,
+        usd_value: to_decimal(current_price),
         volume_24h_usd: to_decimal(market_data["volume"]),
         image_url: image_url
       }
     ]
   end
 
+  @impl Source
+  def format_data(%{"data" => data}) when is_list(data) and length(data) >= 1000 do
+    chain = chain()
+
+    Enum.reduce(data, [], fn item, acc ->
+      case item do
+        %{"blockchains" => blockchains, "contracts" => contracts}
+        when is_list(blockchains) and is_list(contracts) ->
+          with {:ok, index} <- find_contract_index(blockchains, chain),
+               {:ok, contract} <- get_contract(contracts, index),
+               {:ok, token_contract_hash} <- cast_contract_hash(contract) do
```

Comment:
Would you mind rewrite it as follows to avoid creating other functions?
```suggestion
          with index when not is_nil(index) <- Enum.find_index(blockchains, fn bc -> bc == chain end),
                   contract when not is_nil(contract) <- Enum.at(contracts, index),
                   {:ok, token_contract_hash} <- Hash.Address.cast(contract) do
```

Reply:
That's what I wanted to do in the first place, but "mix credo" return an issue:

> Function body is nested too deep (max depth is 2, was 3).

Is that a problem?